data_structures/linked-list/linked_list.py

```diff
@@ -62,6 +62,8 @@ def insert_before(self, val, newVal):
         """Insert a new node before the matching value."""
         search = self.head
         previous = None
+        if type(val) is not int:
+            raise Exception('Please enter an integer.')
```

Comment:
You should only ever raise specific exceptions, never the base, generic `Exception`. In this particular case, the user has given a value of the incorrect type, so this should probably be a `TypeError`.